### DISCUSSION TITLE
BonVision - multiple displays with Normalized View (logic?)

### DISCUSSION THREAD

**Discussion by preinagel** [bonsai-rx/bonsai#1517] [Q&A] [answered]
(New user trying to get started; assume almost no knowledge other than going through the available tutorials online, and searching the forum for this issue.)

My goal is to display visual stimuli on both of two identical display monitors. In CreateWindow I selected DisplayDevice = First, set WindowStyle WindowBorder to "Hidden" and Width parameter to twice the pixel width of one monitor, expecting that window to then fill the First and Second displays. (I am using a Windows 11 PC that supports three displays, and using "Primary" display for the console). Instead, this window covers the first display and half of the 2nd one. I have to pick Width= 3x pixel width of one monitor to fill the two screens (and this actually spills over by a few pixels into the third i.e. Primary display). Is there a way to make sense of how this is behaving? Is there a preferred way to control two displays?

Then, with this window that fills the First and Second displays, I expected that if I used RenderFrame with NormalizedView, stimulus positions in this entire window would be specified from -1 to +1 left to right, so 0 would be at the border between the two displays,  -0.5 in the middle of the left display and +0.5 in the middle of the right display. Not so. Instead, 0 remains at the center of the First display. Moreoever, if I want to center a grating on the left edge of the left display I have to use LocationX= -1.75 (not -1). The edge straddling the two displays is +1.75, center of the 2nd display is +3.5 etc.  Is there a way to understand this? I can make it work, obviously, but I feel like I am missing some core logic about NormalizedView.

**Comment by glopesdev** [comment, marked as answer]
@preinagel The OpenTK shader window unfortunately does not deal too well with OS scale factors. If the coordinates are not adding up I would suspect that if you go to windows Display Settings > Scale it will be set to a value other than 100%. That should in theory be the scale factor you need to take into account when setting the numbers on the window size and location. However, if the monitors have different scale factors it can be tricky to understand which one is actually being used.

This is also part of the reason why we usually recommend trying to merge all external displays into a single "virtual" screen using NVidia Mosaic or Quadro drivers. That way you can then just full-screen the window into the right display and it will be correct. This has the further advantage that the GPU might support hardware frame-sync so that buffer swaps and refresh of both screens will be physically synchronized (which might not be the case when using just extended desktop).

Re. second question about the `NormalizedView` node, there is actually a misconception around this node which is probably down to our own misleading documentation. The only coordinate which is normalized is Y axis, from -1 to 1. The horizontal X axis is actually set according to the aspect ratio of the window size, which is needed when you want circles to render as circles or images to keep their original proportions without stretching. If this is not important and you want to force [-1, 1] on all axes, you should use `OrthographicView` instead and set the boundaries to [Bottom: -1, Left: -1, Right: 1, Top: 1].
